posthog/temporal/tests/batch_exports/test_logger.py

```diff
@@ -0,0 +1,408 @@
+import asyncio
+import dataclasses
+import datetime as dt
+import json
+import random
+import time
+import uuid
+
+import aiokafka
+import freezegun
+import pytest
+import pytest_asyncio
+import structlog
+import temporalio.activity
+import temporalio.testing
+from django.conf import settings
+
+from posthog.clickhouse.client import sync_execute
+from posthog.clickhouse.log_entries import (
+    KAFKA_LOG_ENTRIES_TABLE_SQL,
+    LOG_ENTRIES_TABLE,
+    LOG_ENTRIES_TABLE_MV_SQL,
+    TRUNCATE_LOG_ENTRIES_TABLE_SQL,
+)
+from posthog.kafka_client.topics import KAFKA_LOG_ENTRIES
+from posthog.temporal.workflows.logger import bind_batch_exports_logger, configure_logger
+
+pytestmark = pytest.mark.asyncio
+
+
+class LogCapture:
+    """A test StructLog processor to capture logs."""
+
+    def __init__(self):
+        self.entries = []
+
+    def __call__(self, logger, method_name, event_dict):
+        """Append event_dict to entries and drop the log."""
+        self.entries.append(event_dict)
+        raise structlog.DropEvent()
+
+
+@pytest.fixture()
+def log_capture():
+    """Return a LogCapture processor for inspection in tests."""
+    return LogCapture()
+
+
+class QueueCapture(asyncio.Queue):
+    """A test asyncio.Queue that captures items that we put into it."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.entries = []
+
+    def put_nowait(self, item):
+        """Append item to entries and delegate to asyncio.Queue."""
+        self.entries.append(item)
+        super().put_nowait(item)
+
+
+@pytest_asyncio.fixture()
+async def queue():
+    """Return a QueueCapture queue for inspection in tests."""
+    queue = QueueCapture(maxsize=-1)
+
+    yield queue
+
+
+class CaptureKafkaProducer:
+    """A test aiokafka.AIOKafkaProducer that captures calls to send_and_wait."""
+
+    def __init__(self, *args, **kwargs):
+        self.entries = []
+        self._producer: None | aiokafka.AIOKafkaProducer = None
+
+    @property
+    def producer(self) -> aiokafka.AIOKafkaProducer:
+        if self._producer is None:
+            self._producer = aiokafka.AIOKafkaProducer(
+                bootstrap_servers=settings.KAFKA_HOSTS + ["localhost:9092"],
```

Comment:
Not including extra hosts here making CI tests fail for some reason.

Reply:
Huh, weird, it should be `["kafka:9092"]` which works in CI for other Kafka clients.